modules/practice_adapter.js

```diff
@@ -300,6 +300,13 @@ class PracticeAdapter extends EventEmitter {
 
   async postLeaderboard (leaderboard) {
     if (this._config.get('postLeaderboardOnReset')) {
+      // First unpin any old leaderboards
+      // Assume that all pinned messages in the announcementChannel by the bot are leaderboards
+      this._announcementsChannel.messages.fetchPinned()
+        .then(pinned => pinned.filter(msg => msg.author === this._client.user))
```

Comment:
Suggestion: `pinned.filter(...).forEach(...)`

The filter and forEach calls aren't async.

Reply:
Oh, yes, right. I'll adjust that.